File: cpp/Candies.cpp
#include <bits/stdc++.h>
using namespace std;


int main() {
    int n;
    cin >> n;
    vector<int> arr(n);
    for (int i = 0; i < n; ++i) {
        cin >> arr[i];
    }
    int value = 1, ans = 0;
    if (n >= 2 && arr[1] < arr[0])
        value = 2;
    for (int i = 0; i < n; ++i) {
        if (i == 0) {
            ans = value;
        }
        else {
            if (arr[i] > arr[i - 1]) {
                value += 1;
                ans += value;
            }
            else {
                value = 1;
                for (int j = i; j < n - 1; ++j){
                    if (arr[j] > arr[j + 1]) {
                        value += 1;
                    }
                    else
                        break;
                }
                ans += value;
            }
        }
    }
    cout << ans << endl;
    return 0;
}

File: python/Candies.py
arr = []
for _ in range(int(input())):
    arr.append(int(input()))

ans = [0] * len(arr)
ans[0] = 1
try:
    if arr[1] < arr[0]:
        i = 0
        while arr[i + 1] < arr[i]:
            ans[0] += 1
            i += 1
except :
    pass
print("ans[0]", ans[0])
for i in range(len(arr)):
    if i == 0:
        continue
    if arr[i] > arr[i - 1]:
        ans[i] = ans[i - 1] + 1
    else:
        ans[i] = 1
        j = i
        try:
            while arr[j] > arr[j + 1]:
                ans[i] += 1
                j += 1
        except:
            pass
        if arr[i] < arr[i - 1] and ans[i - 1] <= ans[i]:
            ans[i - 1] = ans[i] + 1

print(arr)
print(ans)
print(sum(ans))
